util/rowDecoder/decoder.go

```diff
@@ -88,17 +120,21 @@ func (rd *RowDecoder) DecodeAndEvalRowWithMap(ctx sessionctx.Context, handle kv.
 	if err != nil {
 		return nil, err
 	}
-	if !rd.haveGenColumn {
-		return row, nil
-	}
-
+	pkCols := tables.TryGetCommonPkColumnIds(tbl)
```

Comment:
The RowDecoder already has a table field, we show create pkCols in NewRowDecoder.